diff --git a/src/execle_open.c b/src/execle_open.c
index 52872aa..6d2821c 100644
--- a/src/execle_open.c
+++ b/src/execle_open.c
@@ -21,6 +21,7 @@
 #include <stdio.h>
 #include <sys/wait.h>
 #include <unistd.h>
+#include <stdlib.h>
 
 static pid_t pid;
 
@@ -32,6 +33,7 @@ execve_close (FILE *f)
 
   if (f != NULL)
     fclose (f);
+    
   while ((p = waitpid (pid, &status, 0)) == -1 && errno == EINTR);
   if (p == -1 || ! WIFEXITED (status))
     return -1;
@@ -50,7 +52,7 @@ execve_open (const char *path, char *const argv[], char *const envp[])
       return NULL;
     }
 
-  switch (vfork ())
+  switch (pid=fork ())
     {
     case -1:
       error (0, errno, "Could not run %s", path);
@@ -63,7 +65,9 @@ execve_open (const char *path, char *const argv[], char *const envp[])
 	  close (p[1]);
 	}
       dup2 (1, 2);
-      execve (path, argv, envp);
+      while (*envp) 
+        putenv(*envp++);
+      execv (path, argv);
       _exit (127);
     }
 
diff --git a/src/gather.c b/src/gather.c
index af4cfa5..80731e1 100644
--- a/src/gather.c
+++ b/src/gather.c
@@ -208,7 +208,8 @@ gather_deps (DSO *dso, struct prelink_entry *ent)
 		      goto error_out;
 		    }
 		}
-	      error (0, 0, "%s: Could not parse `%s'", ent->filename, line);
+	      /* This is not an error situation: error (0, 0, "%s: Could7 not parse `%s'", ent->filename, line); */
+	      continue;
 	    }
 	  goto error_out;
 	}
diff --git a/src/get.c b/src/get.c
index fe32deb..cbcd1c9 100644
--- a/src/get.c
+++ b/src/get.c
@@ -30,6 +30,8 @@ int
 is_ldso_soname (const char *soname)
 {
   if (! strcmp (soname, "ld-linux.so.2")
+      || ! strcmp (soname, "ld-linux.so.3")
+      || ! strcmp (soname, "ld-linux-armhf.so.3")
       || ! strcmp (soname, "ld.so.1")
       || ! strcmp (soname, "ld-linux-ia64.so.2")
       || ! strcmp (soname, "ld-linux-x86-64.so.2")
diff --git a/testsuite/movelibs.sh b/testsuite/movelibs.sh
index 4357ee8..08d3160 100755
--- a/testsuite/movelibs.sh
+++ b/testsuite/movelibs.sh
@@ -6,6 +6,7 @@ $CXX -o movelibs movelibs.C
 > syslib.list
 > syslnk.list
 for i in `LD_TRACE_PRELINKING=1 LD_WARN= LD_TRACE_LOADED_OBJECTS=1 ./movelibs \
+	  | grep " => " \
 	  | awk '$1 !~ /^\.\/movelibs/ { print $3 } '`; do
   k=`basename $i`
   if [ -L $i ]; then
